src/components/PostDetails.tsx

```diff
@@ -2,104 +2,42 @@ import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
 
-import * as commentsApi from '../api/comments';
-
-import { Post } from '../types/Post';
-import { Comment, CommentData } from '../types/Comment';
-
-type Props = {
-  post: Post;
-};
-
-export const PostDetails: React.FC<Props> = ({ post }) => {
-  const [comments, setComments] = useState<Comment[]>([]);
-  const [loaded, setLoaded] = useState(false);
-  const [hasError, setError] = useState(false);
+import { useAppDispatch, useAppSelector } from '../app/hooks';
+import {
+  fetchComments,
+  initRemoveComment,
+  removeComment,
+} from '../features/commentSlice';
+
+export const PostDetails: React.FC = () => {
+  const dispatch = useAppDispatch();
+  const post = useAppSelector(state => state.posts.post);
+  const comments = useAppSelector(state => state.comments.comments);
+  const loaded = useAppSelector(state => state.comments.loaded);
+  const hasError = useAppSelector(state => state.comments.error);
```

Comment:
```suggestion
  const { comments, loaded, error } = useAppSelector(state => state.comments);
```
select state comments and destructure all fields that you need